framework/render/tasks.py

```diff
@@ -39,8 +40,45 @@ def render_mfr_error(err):
         """.format(**locals())
 
 
+# TODO only allow one task at a time
+@app.task(ignore_result=True, timeout=settings.MFR_TIMEOUT)
+def _build_rendered_html(download_url, cache_path, temp_path):
+    """
+    :param str file_path: Full path to raw file on disk
+    :param str cache_dir: Folder to store cached file in
+    :param str cache_file_name: Name of cached file
+    :param str download_url: External download URL
+    """
+    try:
+        ensure_path(os.path.split(temp_path)[0])
+        urllib.urlretrieve(download_url, temp_path)
+    except:
+        # TODO Log to sentry here
+        return
+
+    with codecs.open(temp_path) as file_pointer:
+
+        # Build path to cached content
+        # Note: Ensures that cache directories have the same owner as the files
+        # inside them
+        ensure_path(os.path.split(cache_path)[0])
+
+        with codecs.open(cache_path, 'w', 'utf-8') as write_file_pointer:
+            # Render file
+            try:
+                render_result = mfr.render(file_pointer, src=download_url)
+            except MFRError as err:
+                rendered = render_mfr_error(err)
+            else:
+                rendered = _build_html(render_result)
+
+            # Cache rendered content
+            write_file_pointer.write(rendered)
+
+    os.remove(temp_path)
+
 @app.task(ignore_result=True, timeout=settings.MFR_TIMEOUT)
-def _build_rendered_html(file_path, cache_dir, cache_file_name, download_url):
+def _old_build_rendered_html(file_path, cache_dir, cache_file_name, download_url):
```

Comment:
Could you document the difference between `_build_rendered_html` and `_old_build_rendered_html` and explain why we have both? Do we need both?